### ISSUE TITLE
Netconf regression i.e. 2.5.0 vs 2.4.0 (i.e. netconf_get, netconf_config) with  netconf_os as 'default' 

### ISSUE THREAD

**Issue by Golvanig** [ansible-collections/ansible.netcommon#366]
##### SUMMARY
I have been  using ansible for a year to configure my NETCONF 'Ekinops' devices implementing standard NETCONF features, with **netconf_get** and of course **netconf_config**.

It was perfectly working until I updated Ansible, related Collections, nccllient.
I tried on a fresh Linux, and got the same issue.

##### ISSUE TYPE
Now when I launch my playbooks I do have:
...
The error was: ansible.module_utils.connection.ConnectionError: Method not found
...

##### COMPONENT NAME
ansible.netcommon             2.5.0  
ansible.utils                 2.4.3 

ncclient                                         0.6.12

##### ANSIBLE VERSION

```
ansible [core 2.12.1]
  config file = /home/thierry/int/ekinops/ansible.cfg
  configured module search path = ['/home/thierry/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/thierry/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/thierry/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/thierry/.local/bin/ansible
  python version = 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
  jinja version = 3.0.3
  libyaml = True
```

##### COLLECTION VERSION
```
ansible.netcommon             2.5.0  
ansible.utils                 2.4.3 

```
##### CONFIGURATION
```
CACHE_PLUGIN(/home/thierry/int/ekinops/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/home/thierry/int/ekinops/ansible.cfg) = facts.json
CACHE_PLUGIN_TIMEOUT(/home/thierry/int/ekinops/ansible.cfg) = 86400
DEFAULT_GATHERING(/home/thierry/int/ekinops/ansible.cfg) = smart
DEFAULT_GATHER_SUBSET(/home/thierry/int/ekinops/ansible.cfg) = ['all']
DEFAULT_HOST_LIST(/home/thierry/int/ekinops/ansible.cfg) = ['/home/thierry/int/ekinops/hosts.yml']
DEFAULT_TIMEOUT(/home/thierry/int/ekinops/ansible.cfg) = 30
HOST_KEY_CHECKING(/home/thierry/int/ekinops/ansible.cfg) = False

```

##### OS / ENVIRONMENT
uname -a
Linux xxxxxx 5.15.0-3-amd64 #1 SMP Debian 5.15.15-1 (2022-01-18) x86_64 GNU/Linux

##### STEPS TO REPRODUCE
launch playbook to retrieve device configuration with 'netconf_get'
"myplaybool.yml"
---
- hosts: cpe
  gather_facts: false

  tasks:
    - name: fetch initial state
      connection: ansible.netcommon.netconf
      netconf_get:
        display: json
        source: running
      register: result

with "hosts.yml" as:
---
cpe:
  hosts:
    cpe01:
      ansible_host: loc.cpe.conf

all:
  vars:
    ansible_connection: ansible.netcommon.netconf
    ncclient_device_handler: "default"
    ansible_user: admin
    ansible_password: admin
  children:
    cpe:

```
**ncclient** is working perfectly well. With for instance test.py as:
#!/usr/bin/env python

from ncclient import manager
import sys

HOST = 'loc.cpe.conf'
PORT = 830
USER = 'admin'
PASS = 'admin'

def main():
    """Main method that prints netconf capabilities of remote device."""
    with manager.connect(host=HOST, port=PORT, username=USER, password=PASS,
                         hostkey_verify=False, device_params={'name': 'default'},
                         look_for_keys=False, allow_agent=False) as m:

        # print all NETCONF capabilities
        print('***Here are the Remote Devices Capabilities***')
        for capability in m.server_capabilities:
            print(capability)

if __name__ == '__main__':
    sys.exit(main())

-------------------------------------------------
get: with "python test.py" 
***Here are the Remote Devices Capabilities***
urn:ietf:params:netconf:base:1.0
urn:ietf:params:netconf:base:1.1
urn:ietf:params:netconf:capability:candidate:1.0
---etc---
...


I can connect with ssh admin@loc.cpe.conf -p 830 -s netconf 
I can also use perfectly netconf-console2

##### EXPECTED RESULTS
Should be able to retrieve the running configuration as before...


##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->
<loc.cpe.conf> EXEC /bin/sh -c 'rm -f -r /home/thierry/.ansible/tmp/ansible-local-46972fwlsl826/ansible-tmp-1643216036.4095333-46976-41602913194032/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/home/thierry/.ansible/tmp/ansible-local-46972fwlsl826/ansible-tmp-1643216036.4095333-46976-41602913194032/AnsiballZ_netconf_get.py", line 107, in <module>
    _ansiballz_main()
  File "/home/thierry/.ansible/tmp/ansible-local-46972fwlsl826/ansible-tmp-1643216036.4095333-46976-41602913194032/AnsiballZ_netconf_get.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/thierry/.ansible/tmp/ansible-local-46972fwlsl826/ansible-tmp-1643216036.4095333-46976-41602913194032/AnsiballZ_netconf_get.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.ansible.netcommon.plugins.modules.netconf_get', init_globals=dict(_module_fqn='ansible_collections.ansible.netcommon.plugins.modules.netconf_get', _modlib_path=modlib_path),
  File "/usr/lib/python3.10/runpy.py", line 209, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible_collections/ansible/netcommon/plugins/modules/netconf_get.py", line 419, in <module>
  File "/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible_collections/ansible/netcommon/plugins/modules/netconf_get.py", line 310, in main
  File "/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/netconf/netconf.py", line 60, in get_capabilities
  File "/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible/module_utils/connection.py", line 200, in __rpc__
ansible.module_utils.connection.ConnectionError: Method not found
fatal: [cpe01]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/thierry/.ansible/tmp/ansible-local-46972fwlsl826/ansible-tmp-1643216036.4095333-46976-41602913194032/AnsiballZ_netconf_get.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/thierry/.ansible/tmp/ansible-local-46972fwlsl826/ansible-tmp-1643216036.4095333-46976-41602913194032/AnsiballZ_netconf_get.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/thierry/.ansible/tmp/ansible-local-46972fwlsl826/ansible-tmp-1643216036.4095333-46976-41602913194032/AnsiballZ_netconf_get.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.ansible.netcommon.plugins.modules.netconf_get', init_globals=dict(_module_fqn='ansible_collections.ansible.netcommon.plugins.modules.netconf_get', _modlib_path=modlib_path),\n  File \"/usr/lib/python3.10/runpy.py\", line 209, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible_collections/ansible/netcommon/plugins/modules/netconf_get.py\", line 419, in <module>\n  File \"/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible_collections/ansible/netcommon/plugins/modules/netconf_get.py\", line 310, in main\n  File \"/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/netconf/netconf.py\", line 60, in get_capabilities\n  File \"/tmp/ansible_netconf_get_payload_z59ztzi4/ansible_netconf_get_payload.zip/ansible/module_utils/connection.py\", line 200, in __rpc__\nansible.module_utils.connection.ConnectionError: Method not found\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

<!--- Paste verbatim command output between quotes -->

It seems that netconf is not even trying to connect to my device.
When I fake the credentials, there is no complain. However I guess that there is something wrong before...

I tried to dig into the source to understand, but this is a quite sophisticated work which does not accept new comers easily ;)

I need help. Thanks in advance

**Comment by Golvanig** [comment]
The test is OK when I use 2.4.0 with  "**ansible-galaxy collection install ansible.netcommon:==2.4.0**"
and of course after removing of 2.5.0

**Comment by jksmth** [comment]
Hi, just to add a +1. I have the same error with this version and network_os = default

I have a number of non vendor specific devices that support netconf.

**Comment by jksmth** [comment]
The regression seems to be caused by #334

**Comment by jksmth** [comment]
I believe this will be fixed by #358

**Comment by Qalthos** [comment]
I'm closing this as a duplicate of #356, as the problem and solution have both already been identified. Thanks for the report!
